viper.go

```diff
@@ -1452,29 +1455,30 @@ func (v *Viper) SetDefault(key string, value interface{}) {
 	v.lock.Unlock()
 }
 
+// SetDefault sets the default value for this key.
+// SetDefault is case-insensitive for a key.
+// Default only used when no value is provided by the user via flag, config or ENV.
+func SetDefault(key string, value interface{}) { v.SetDefault(key, value) }
+func (v *Viper) SetDefault(key string, value interface{}) {
+	v.setInMap(key, value, v.defaults)
+}
+
+// SetType sets the type for this key.
+// This type is used for type conversions, e.g. a slice from an env var
+// This function allows the default to be nil while still enabling those type conversions configured
+// through SetTypeByDefaultValue
+func SetType(key string, t interface{}) { v.SetType(key, t) }
+func (v *Viper) SetType(key string, t interface{}) {
+	v.setInMap(key, t, v.types)
```

Comment:
Is it intentional that the cache will be cleared here?

Reply:
I think it should be as the wrong type might be in the cache

Reply:
ok, soudns good!